### PR TITLE
[qt6] Fixes linguisttools

### PR DESCRIPTION
**Pull request by mjvankampen**
Specify library name and version:  **qt/6.x.x**

Linguist Tools is not a library but a collection of targets (which refer to binaries) and needs to be added in a different manner. Now you will find that things will start and try linking Qt6LinguistTools.lib.

Ref. #6877
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/qt/6.x.x/conanfile.py [hunk 1]
```diff
@@ -788,7 +788,6 @@ def _create_plugin(pluginname, libname, type, requires):
             _create_module("QuickTest", ["Test"])
 
         if self.options.qttools and self.options.gui and self.options.widgets:
-            _create_module("LinguistTools", ["Gui", "Widgets"])
             _create_module("UiPlugin", ["Gui", "Widgets"])
```
Comment (ericLemanissier) [inline recipes/qt/6.x.x/conanfile.py:791]:
```suggestion
            self.cpp_info.components["qtLinguistTools"].names["cmake_find_package"] = "LinguistTools"
            self.cpp_info.components["qtLinguistTools"].names["cmake_find_package_multi"] = "LinguistTools"
            _create_module("UiPlugin", ["Gui", "Widgets"])
```
This should allow to create a LinguistTools module, which gives access to qt6_create_translation  and qt6_add_translation cmake macros

**Comment by ericLemanissier** [comment]
@Minimonium would you have a chance to test if this works for you ?

**Comment by Minimonium** [comment]
Looks good! Yes, it's not a proper library, I was not familiar enough with the structure of `_create_module`, assumed it's the way to append components. That solution looks better.

File: recipes/qt/6.x.x/conanfile.py (hunk 1 above)
Comment (ericLemanissier) [reply recipes/qt/6.x.x/conanfile.py:793]: We should probably do the same for Platform above:
replace
```
        _create_module("Platform")
        self.cpp_info.components["qtPlatform"].libs = [] # this is a collection of abstract classes, so this is header-only
        self.cpp_info.components["qtPlatform"].libdirs = []
```
with
```
            self.cpp_info.components["qtPlatform"].names["cmake_find_package"] = "Platform"
            self.cpp_info.components["qtPlatform"].names["cmake_find_package_multi"] = "Platform"
```

**Comment by AndreyMlashkin** [comment]
Is it possible to add a translation test to a test_package?
Similar way as we have a test for qrc (resource files)

**Comment by ericLemanissier** [comment]
sure, don't hesitate to do so

**Review by AndreyMlashkin — DISMISSED** [review]

**Review by ericLemanissier — CHANGES_REQUESTED** [review]
@mjvankampen 
We should probably do the same for Platform:
replace
```
        _create_module("Platform")
        self.cpp_info.components["qtPlatform"].libs = [] # this is a collection of abstract classes, so this is header-only
        self.cpp_info.components["qtPlatform"].libdirs = []
```
with
```
            self.cpp_info.components["qtPlatform"].names["cmake_find_package"] = "Platform"
            self.cpp_info.components["qtPlatform"].names["cmake_find_package_multi"] = "Platform"
```

**Review by SSE4 — DISMISSED** [review]

**Comment by AndreyMlashkin** [comment]
Could you also cherry-pick my bump versions commit?
https://github.com/conan-io/conan-center-index/pull/6835/commits/2efb15a715e2bc4d340bed36e22bcc826cc68563

**Comment by daravi** [comment]
Hi just wondering if this also affects 5.x.x recipe?

File: recipes/qt/6.x.x/conanfile.py [hunk 2]
```diff
@@ -788,7 +788,8 @@ def _create_plugin(pluginname, libname, type, requires):
             _create_module("QuickTest", ["Test"])
 
         if self.options.qttools and self.options.gui and self.options.widgets:
-            _create_module("LinguistTools", ["Gui", "Widgets"])
+            self.cpp_info.components["qtLinguistTools"].names["cmake_find_package"] = "LinguistTools"
+            self.cpp_info.components["qtLinguistTools"].names["cmake_find_package_multi"] = "LinguistTools"
```
Comment (prince-chrismc) [inline recipes/qt/6.x.x/conanfile.py:792]: should it link to threads like https://github.com/conan-io/conan-center-index/pull/6877#issuecomment-900994113? or is that handled else where?

Comment (mjvankampen) [reply recipes/qt/6.x.x/conanfile.py:791]: The comment is for the platform module which I think is platform specific. I would say we fix linguist first and worry about platform later as I'm not sure how complex that will be to tackle.

**Comment by mjvankampen** [comment]
> 
> 
> Hi just wondering if this also affects 5.x.x recipe?

This PR does not change the 5.x.x recipe. As for linguist and 5.x.x I do not know. I don't use either so I'm afraid I can't help you with that.

**Review by ericLemanissier — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
Looking forward to see these changes in master :D

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**